letsencrypt/cli.py

```diff
@@ -87,6 +87,48 @@
 """
 
 
+# These argparse parameters should be removed when detecting defaults.
+ARGPARSE_PARAMS_TO_REMOVE = ("const", "nargs", "type",)
+
+
+# These sets are used when to help detect options set by the user.
+EXIT_ACTIONS = set(("help", "version",))
+
+
+ZERO_ARG_ACTIONS = set(("store_const", "store_true",
+                        "store_false", "append_const", "count",))
```

Comment:
Stylistically, we should probably move all of this default detection machinery to a different file (though that will require inheriting from `HelpfulArgumentParser` rather than tweaking it) or at least a clearly marked section of this file, because it's probably confusing that these variables are scattered amongst the regular CLI parsing case.

But that can be another PR rather than this one.